### ISSUE TITLE
CI issues after flake8 has been bumped to 6.0.0

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [arthurdejong/python-stdnum#342]
https://github.com/PyCQA/flake8/issues/1760

* **'choice' is not callable**
  <details>
  <summary>partial error backtrace from CI logs</summary>

  ```
     File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8/main/application.py", line 165, in initialize
      self.plugins, self.options = parse_args(argv)
                                   ^^^^^^^^^^^^^^^^
    File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8/options/parse_args.py", line 51, in parse_args
      option_manager.register_plugins(plugins)
    File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8/options/manager.py", line 259, in register_plugins
      add_options(self)
    File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8_quotes/__init__.py", line 109, in add_options
      cls._register_opt(parser, '--quotes', action='store',
    File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8_quotes/__init__.py", line 99, in _register_opt
      parser.add_option(*args, **kwargs)
    File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8/options/manager.py", line 281, in add_option
      self._current_group.add_argument(*option_args, **option_kwargs)
    File "/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/argparse.py", line 1448, in add_argument
      raise ValueError('%r is not callable' % (type_func,))
  ValueError: 'choice' is not callable
  ERROR: InvocationError for command /home/runner/work/python-stdnum/python-stdnum/.tox/flake8/bin/flake8 stdnum tests update setup.py (exited with code 1)
  ```
  </details>
  
  Plugin [flake8-quotes](https://github.com/zheller/flake8-quotes) is incompatible with flake8 v6, see https://github.com/zheller/flake8-quotes/issues/110. Either we disable [flake8-quotes](https://github.com/zheller/flake8-quotes) in CI, or we pin flake8 to v5.
* **ValueError: Error code '#' supplied to 'ignore' option does not match '^[A-Z]{1,3}[0-9]{0,3}$'**
  <details>
  <summary>partial error backtrace from CI logs</summary>

  ```
    File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8/main/application.py", line 165, in initialize
      self.plugins, self.options = parse_args(argv)
                                   ^^^^^^^^^^^^^^^^
    File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8/options/parse_args.py", line 53, in parse_args
      opts = aggregator.aggregate_options(option_manager, cfg, cfg_dir, rest)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8/options/aggregator.py", line 30, in aggregate_options
      parsed_config = config.parse_config(manager, cfg, cfg_dir)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/runner/work/python-stdnum/python-stdnum/.tox/flake8/lib/python3.11/site-packages/flake8/options/config.py", line 131, in parse_config
      raise ValueError(
  ValueError: Error code '#' supplied to 'ignore' option does not match '^[A-Z]{1,3}[0-9]{0,3}$'
  ERROR: InvocationError for command /home/runner/work/python-stdnum/python-stdnum/.tox/flake8/bin/flake8 stdnum tests update setup.py (exited with code 1)
  ```
  </details>

    The syntax of [`setup.cfg`](https://github.com/arthurdejong/python-stdnum/blob/master/setup.cfg) is incorrect. While pinning flake8 will silence this error, we can easily fix the root cause.

**Comment by arthurdejong** [comment]
In 7a91a98 flake8 was pinned to avoid version 6.0 but since then I've not seem much movement in flake8. In #410 it was suggested to use `ruff` in place of `flake8` which seems promising (but it is a bit weird to have a tool written in Rust and flake8 performance has never been a thing that I thought was a problem).

Anyway, I'm closing this issue due to the fix in 7a91a98 but if there are other concerns feel free to re-open or add feedback to any of the other issues.

**Comment by DimitriPapadopoulos** [comment]
It looks like [flake8-quotes](https://github.com/zheller/flake8-quotes) has been updated to support flake8 6.0:
https://github.com/zheller/flake8-quotes/pull/111

An alternative to https://github.com/arthurdejong/python-stdnum/commit/7a91a98d754a6d80537ccce06ab418a779342a94 would be to require version 3.3.2 released on 19 December 2022.
